packages/kit/src/runtime/server/respond.js

```diff
@@ -508,11 +508,9 @@ export async function respond(request, options, manifest, state) {
 			}
 
 			if (state.error && event.isSubRequest) {
-				return await fetch(request, {
-					headers: {
-						'x-sveltekit-error': 'true'
-					}
-				});
+				const headers = new Headers(request.headers);
```

Comment:
I'm fine with not having a test for this, but we should have a short comment here explaining why we do this

Reply:
added a comment